### PR TITLE
html-css-popup

### PR DESCRIPTION
**Pull request by YuliiaDikun**
### HTML і CSS практика: Hooli-style Popup

[Demo](https://yuliiadikun.github.io/Popup/) |
[Code base](https://github.com/YuliiaDikun/Popup)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/dikun/html-css-popup/style.css [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    border: 1px solid #fff;
+    border-radius: 5px;    
+}
+
+.input__popup-even:focus + .label__popup-even {
+    border: 1px solid #fff;
+    border-radius: 5px;    
+}
+
+@media (max-width: 430px) {
```
Comment (A-Ostrovnyy) [inline submissions/dikun/html-css-popup/style.css:219]: :fire:

File: submissions/dikun/html-css-popup/index.html [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                            </li>
+                            <li class="popup__item">
+                                <a href="#" class="popup__link">
+                                    <img class="popup__img" src="icons/apps.png" alt="">
+                                    <p>Apps</p>
+                                </a>
+                            </li>
+                            <li class="popup__item">
+                                <a href="#" class="popup__link">
+                                    <img class="popup__img" src="icons/documents.png" alt="Documents">
```
Comment (A-Ostrovnyy) [inline submissions/dikun/html-css-popup/index.html:96]: [Read more](https://htmlacademy.ru/blog/articles/alt-text) about how to populate `alt` attribute. Sorry for russian but it's a good article.

File: submissions/dikun/html-css-popup/index.html [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                                    </li>
+                                    <li class="popup__item">
+                                        <a href="#" class="popup__link">
+                                            <img class="popup__img" src="icons/blogs.png" alt="">
+                                            <p>Blogs</p>
+                                        </a>
+                                    </li>                                     
+                                </ul>
+                                <input type="checkbox" id="check-even-more" class="input__popup-even">
+                                    <label for="check-even-more" class="label__popup-even">
```
Comment (A-Ostrovnyy) [inline submissions/dikun/html-css-popup/index.html:125]: Fix indentation.

File: submissions/dikun/html-css-popup/index.html [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+                                    <a href="https://github.com/kottans/frontend/blob/2022_UA/tasks/html-css-popup.md" target="_blank">Even more</a>
```
Comment (A-Ostrovnyy) [inline submissions/dikun/html-css-popup/index.html:126]: Avoid including one interactive element inside another.
When user get focused on button "Even more" and presses key Tab focus must go to the next interactive element, in your case it goes to text. Avoid this situation in the future.

File: submissions/dikun/html-css-popup/index.html [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        <nav class="header__navigation">
+            <ul class="navigation__items">
+                <li class="navigation__item">
+                    <a href="#" class="navigation__link">Mail</a>
+                </li>
+                <li class="navigation__item">
+                    <a href="#" class="navigation__link">Images</a>
+                </li>
+                <li class="navigation__item">
+                    <input type="checkbox" id="check" class="input__popup">
```
Comment (A-Ostrovnyy) [inline submissions/dikun/html-css-popup/index.html:22]: Check your BEM.

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Awesome!
See my comments below.

File: submissions/dikun/html-css-popup/index.html [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+        <nav class="header__navigation">
+            <ul class="navigation__items">
+                <li class="navigation__item">
+                    <a href="#" class="navigation__link">Mail</a>
+                </li>
+                <li class="navigation__item">
+                    <a href="#" class="navigation__link">Images</a>
+                </li>
+                <li class="navigation__item popup">
+                    <input type="checkbox" id="check" class="input__popup">
```
Comment (A-Ostrovnyy) [inline submissions/dikun/html-css-popup/index.html:22]: Not fixed yet.
Your input has class `input_popup` which means this is a BEM element of the parent BEM block with class `input`. I can't see BEM block with class `.input`)

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good work!
One more issue left

File: submissions/dikun/html-css-popup/index.html (hunk 6 above)
Comment (YuliiaDikun) [reply submissions/dikun/html-css-popup/index.html:22]: Add `input` class.

File: submissions/dikun/html-css-popup/index.html [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+            <ul class="navigation__items">
+                <li class="navigation__item">
+                    <a href="#" class="navigation__link">Mail</a>
+                </li>
+                <li class="navigation__item">
+                    <a href="#" class="navigation__link">Images</a>
+                </li>
+                <li class="navigation__item input">
+                    <input type="checkbox" id="check" class="input__popup">
+                    <label for="check" class="label__popup">
```
Comment (A-Ostrovnyy) [inline submissions/dikun/html-css-popup/index.html:23]: You should add parent BEM block for this element

File: submissions/dikun/html-css-popup/index.html [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+                       <img src="icons/popup-button.png" alt="" class="label__img">
+                    </label>
+                    <div class="popup__content">
```
Comment (A-Ostrovnyy) [inline submissions/dikun/html-css-popup/index.html:26]: This BEM element should have parent BEM block too)

File: submissions/dikun/html-css-popup/index.html [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+                                    <li class="popup__item">
+                                        <a href="#" class="popup__link">
+                                            <img class="popup__img" src="icons/classroom.png" alt="">
+                                            <p>Classroom</p>
+                                        </a>
+                                    </li>
+                                    <li class="popup__item">
+                                        <a href="#" class="popup__link">
+                                            <img class="popup__img" src="icons/blogs.png" alt="">
+                                            <p>Blogs</p>
```
Comment (A-Ostrovnyy) [inline submissions/dikun/html-css-popup/index.html:120]: Check nesting your markup: [https://caninclude.glitch.me/caninclude?child=p&parent=a](https://caninclude.glitch.me/caninclude?child=p&parent=a)

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Some BEM elements need their BEM blocks. Also, after pushing your changes click re-request review to notify the reviewer.
image

**Comment by YuliiaDikun** [comment]
Please, look again at my BEM structure. Also, I change the `p` tag in the `a` tag on the `span` tag. I hope, it's okay now. Thank you for your time!

**Review by A-Ostrovnyy — APPROVED** [review]
Well done!
Congratulations

**Pull request merged**